util/reggen/field.py

```diff
@@ -61,25 +62,23 @@
 }
 
 
+@dataclass
```

Comment:
I would strongly recommend frozen dataclasses; frozen dataclasses will have their `__hash__` method set automatically. It will also avoid a whole class bugs.

```suggestion
@dataclass(frozen=True)
```

If you really can't, use `unsafe_hash` to have the `__hash__` method implemented.

```suggestion
@dataclass(unsafe_hash=True)
```